fsw/cfe-core/src/es/cfe_es_task.c

```diff
@@ -427,214 +427,210 @@ void CFE_ES_TaskPipe(CFE_SB_MsgPtr_t Msg)
     uint16         CommandCode;
 
     MessageID = CFE_SB_GetMsgId(Msg);
-    switch (MessageID)
+    if (CFE_SB_MsgId_Equal(MessageID, CFE_ES_SEND_HK_MID))
```

Comment:
Why not leave the swtich, and just CFE_SB_MsgIdToValue?  I'd prefer a one line change vs the approach shown here.  Also if CFE_SB_MsgId_Equal ever gets more complex, the approach here seems like it could lead to inefficiencies, and doesn't really solve the bigger design issue of hardcoded values.  I recommend keeping changes simple/minimal until the full MsgId/SB updated design concepts get reviewed and approved by the major stakeholders.

Reply:
A previous version of this change did that, but that implies/requires that the MsgId is always/completely representable as a single integer, which is part of the problem this is trying to address.  Note that extended header fields already exceed 32 bits for the full address range, so even a 32-bit integer msgid is insufficient to fully support current standards.

Admittedly it makes a lot more sense if the literals are also type-correct.  They are still integers here.

Reply:
This was also a reason I asked this question: https://github.com/nasa/cFE/issues/263#issuecomment-612053991 before submitting this pull request.

FWIW, The CFE_SB_MsgIdToValue is also not good for the same reason, but its use should be confined to syslog/events for now.  Someday should consider replacing it with a custom print routine (MsgIdToString maybe) for this purpose.

Reply:
Still sound like an attempt to solve a bigger issue prior to the necessary design reviews, trades, and stakeholder buy-in.  I don't see the issue you reference, MsgId may not always be completely represented as a single integer, documented as an issue and evaluated/discussed prior to this change.   I'd rather this just focus on type correctness (use the conversions where appropriate) for now.  Alternatively, split exposing IsValidMsgId from this change if it's going to require more discussion.

Reply:
Sounds like this is going to be a bigger discussion.  Please split out exposing IsValidMsgId.

Reply:
OK, I can change that back to the other version.  Again, this was why I posed the question before even testing and submitting this pull, I thought we were OK with this.

TBH, Getting rid of the local dispatch routine in favor of a common approach (as noted in https://github.com/nasa/cFE/issues/263#issuecomment-612144320) is the best approach anyway, so neither of these is really a long term/recommended pattern anyway.

Reply:
I should have been more clear "good to go" does not mean approved pull request and ready to merge to master.

Reply:
Understood on that, but this was a case of confirming what to include/not include prior to submission, not approval of a pull request before it was submitted.  

I had preemptively asked specifically about whether to include this switch() change as there is a time cost associated with re-doing and re-validating the pull request again.  I will change it back, but this back-and-forth is precisely what I was hoping to avoid by posing the question on content ahead of time.

Reply:
If you are concerned about the time you are spending on the MsgId type safety work, then postpone it as I originally requested.  It is not a priority this cycle, and any real changes that happen prior to a full design review with stakeholder buy-in are at risk of being wasted work anyways.

Reply:
It is not big time-wise because that version of the change already (mostly) existed in a previous branch.  Its just the time to rebase/merge and re-test.  I do think its critical to get his moving in the right direction WRT MsgId values which is arguably very broken right now so I wouldn't want to drop this entirely.  My frustration was only because  on Friday (I thought) I confirmed that replacing the `switch()` was OK to include and today it was not acceptable.  Nonetheless I'll call this conversation resolved as the 67eaf73 should be OK.